trestle/core/markdown/control_markdown_node.py

```diff
@@ -145,70 +165,51 @@ def _add_child_prose_if_need(part: common.Part, text: str, should_add: bool):
+                        subtree.content.raw_text,
+                        section_heading_type in [
+                            ControlSectionType.control_statement,
+                            ControlSectionType.control_objective,
+                            ControlSectionType.control_guidance
+                        ]
+                    )
                     continue
                 else:
                     i -= 1  # need to revert back one line to properly handle next heading
                     break  # level of the header is above or equal to the current level, subtree is over
-            if is_control_statement:
-                # Read control statement to a part object
-                statement_id = ControlInterface.create_statement_id(tree_context.control_id)
-                content.statement_part = self._create_part_if_needed(
-                    content.statement_part, const.STATEMENT, statement_id
-                )
-                i = self._process_part_line(i, line, lines, content.statement_part)
-                continue
-
-            if is_control_objective:
-                # Read control objective to a part object
-                content.objective_part = self._create_part_if_needed(
-                    content.objective_part, 'objective', f'{tree_context.control_id}_obj'
-                )
-                i = self._process_part_line(i, line, lines, content.objective_part)
-                continue
 
-            if is_control_guidance:
-                # Read control guidance to a part object
-                guidance_id = ControlInterface.strip_to_make_ncname(tree_context.control_id + '_gdn')
-                content.guidance_part = self._create_part_if_needed(
-                    content.guidance_part, control_md_heading_label_ncname, guidance_id
-                )
-                i = self._process_part_line(i, line, lines, content.guidance_part, read_parts=False)
-                continue
-
-            if is_generic_control_part:
-                # Read other control parts to a part objects
-                if not content.other_parts:
-                    if not tree_context.control_id:
-                        raise TrestleError(
-                            'Unexpected error, control id, group and title should be before ## Control parts.'
-                            'However, none was found.'
-                        )
-
-                    part_id = ControlInterface.strip_to_make_ncname(
-                        tree_context.control_id + '_' + control_md_heading_label
-                    )
-                    a_part = common.Part(id=part_id, name=control_md_heading_label_ncname, prose='')
-                    content.other_parts.append(a_part)
-
-                i = self._process_part_line(i, line, lines, content.other_parts[-1], read_parts=False)
+            if section_heading_type not in [ControlSectionType.undefined, ControlSectionType.editable_content]:
+                # Read part
```

Comment:
Should we have debug message or warning for undefined type?  It may represent content the user meant as an important part and it is being dropped - so I think a warning.